pkg/common/cns-lib/volume/util.go

```diff
@@ -56,6 +57,14 @@ func IsDiskAttached(ctx context.Context, vm *cnsvsphere.VirtualMachine, volumeID
 					virtualDevice := device.GetVirtualDevice()
 					if backing, ok := virtualDevice.Backing.(*vimtypes.VirtualDiskFlatVer2BackingInfo); ok {
 						log.Infof("Found diskUUID %s for volume %s on vm %+v", backing.Uuid, volumeID, vm)
+						if checkNVMeController && hasNVMEController(ctx, vmDevices) {
```

Comment:
what if the disk is not attached to the NVME controller and attached non-NVME controller.
We should not assume if VM has NVME controller, then disk is attached to that NVME controller,.

Reply:
Addressed by checking if the disk on the NVMe controller